src/components/ScmRepositoryDetailPage/ScmRepositoryDetailPage.tsx

```diff
@@ -0,0 +1,145 @@
+import { Button, ClipboardCopy, Split, SplitItem, Text, TextContent, TextVariants, Tooltip } from '@patternfly/react-core';
+import { ExternalLinkAltIcon, InfoCircleIcon } from '@patternfly/react-icons';
+import { useEffect } from 'react';
+import { useParams } from 'react-router-dom';
+
+import { PageTitles } from 'common/constants';
+
+import { useServiceContainer } from 'hooks/useServiceContainer';
+import { useTitle } from 'hooks/useTitle';
+
+import { ActionButton } from 'components/ActionButton/ActionButton';
+import { AttributesItems } from 'components/AttributesItems/AttributesItems';
+import { ContentBox } from 'components/ContentBox/ContentBox';
+import { PageLayout } from 'components/PageLayout/PageLayout';
+import { ServiceContainerLoading } from 'components/ServiceContainers/ServiceContainerLoading';
+import { Toolbar } from 'components/Toolbar/Toolbar';
+import { ToolbarItem } from 'components/Toolbar/ToolbarItem';
+
+import * as scmRepositoryApi from 'services/scmRepositoryApi';
+
+export const ScmRepositoryDetailPage = () => {
+  const { scmRepositoryId } = useParams();
+
+  const serviceContainerScmRepository = useServiceContainer(scmRepositoryApi.getScmRepository);
+  const serviceContainerScmRepositoryRunner = serviceContainerScmRepository.run;
+
+  useEffect(() => {
+    serviceContainerScmRepositoryRunner({ serviceData: { id: scmRepositoryId } });
+  }, [serviceContainerScmRepositoryRunner, scmRepositoryId]);
+
+  useTitle(
+    serviceContainerScmRepository.data?.id
+      ? `${serviceContainerScmRepository.data.id} | ${PageTitles.repositories}`
+      : `Error loading ${PageTitles.projectDetail}`
+  );
+
+  const internalScmRepositoryLink = (internalUrl: string) => {
+    /**
+     * Parses internal repo url to Gerrit gitweb link of the project
+     */
+    const parseInternalRepoLink = (url: string) => {
+      let protocol = url.split('://')[0];
```

Comment:
could it be useful to use `URL()` ( https://developer.mozilla.org/en-US/docs/Web/API/URL/URL ) as it contains individual properties like `host` etc.?

Reply:
`URL()` would not be the best for this scenario, since we are dealing with git urls like `git://code.stage.engineering.redhat.com/jboss-fuse/cxf1.git ` instead of an actual url link, and adding `http://` or removing` git://` would be too much away from what we want.